libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -317,53 +313,60 @@ private static JToken EntityResultToInstanceJObject(string text, Result result)
 
         private void TryScoreEntities(string text, RecognizerResult recognizerResult)
         {
-            if (!this.ScoreEntities)
+            if (this._resolver == null)
             {
                 return;
             }
 
-            var results = _resolver.Score(text, LabelType.Entity);
-            recognizerResult.Properties.Add(EntitiesProperty, results);
-
-            if (results.Any())
+            try
             {
-                if (recognizerResult.Entities == null)
-                {
-                    recognizerResult.Entities = new JObject();
-                }
+                var results = _resolver.Score(text, LabelType.Entity);
+                recognizerResult.Properties.Add(EntitiesProperty, results);
```

Comment:
Since we are visiting this codebase and adding some restructuring, it would be nice, if possible, to add some comments and explain what is the json format, what we are trying to extract, and other decisions that are not obvious in the code.

Reply:
@carlosscastro, it's modeled after what's in LuisRecognizer.